sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FoldablePropagationSuite.scala

```diff
@@ -118,14 +118,30 @@ class FoldablePropagationSuite extends PlanTest {
       Seq(
         testRelation.select(Literal(1).as('x), 'a).select('x + 'a),
         testRelation.select(Literal(2).as('x), 'a).select('x + 'a)))
-      .select('x)
```

Comment:
```
  test("Propagate in subqueries of Union queries") {
    val query = Union(
      Seq(
        testRelation.select(Literal(1).as('x), 'a).select('x, 'x + 'a),
        testRelation.select(Literal(2).as('x), 'a).select('x, 'x + 'a)))
      .select('x)
    val optimized = Optimize.execute(query.analyze)
    val correctAnswer = Union(
      Seq(
        testRelation.select(Literal(1).as('x), 'a)
          .select(Literal(1).as('x), (Literal(1).as('x) + 'a).as("(x + a)")),
        testRelation.select(Literal(2).as('x), 'a)
          .select(Literal(2).as('x), (Literal(2).as('x) + 'a).as("(x + a)"))))
      .select('x).analyze

    comparePlans(optimized, correctAnswer)
  }
```

How about this? The `x` (under `Union`) is transformed as the alias of a literal but the top `x` (above `Union`) is not propagated. This can validate the white list does not include `Union`.